components/ratgdo/select/ratgdo_select.cpp

```diff
@@ -0,0 +1,50 @@
+#include "ratgdo_select.h"
+#include "../ratgdo_state.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace ratgdo {
+
+    static const char* const TAG = "ratgdo.select";
+
+    void RATGDOSelect::dump_config()
+    {
+        LOG_SELECT("", "RATGDO Select", this);
+        ESP_LOGCONFIG(TAG, "  Type: Time to close (TTC)");
+    }
+
+    void RATGDOSelect::setup()
+    {
+        this->parent_->subscribe_ttc_seconds([=](uint16_t value) {
+            if (value == 0) {
+                this->publish_state(std::string("Off"));
+            } else if (value == 60) {
+                this->publish_state(std::string("1 Minute"));
+            } else if (value == 300) {
+                this->publish_state(std::string("5 Minutes"));
+            } else if (value == 600) {
+                this->publish_state(std::string("10 Minutes"));
+            }
+        });
+    }
+
+    void RATGDOSelect::control(const std::string &value) 
+    {
+        // if (this->optimistic_)
+             this->publish_state(value);
+
+        if (value == std::string("Off")) {
```

Comment:
Might be less overhead to get the c_str and compare

Reply:
std::string has a compare() so I used that instead.